File: Greatest-Common-Divisor-of-Strings.py
class Solution(object):
    def gcdOfStrings(self, str1, str2):
        if str1 + str2 != str2 + str1:
            return ""

        length1 = len(str1)
        length2 = len(str2)

        while length2:
            length1, length2 = length2, length1 % length2

        return str1[:length1]

File: Greatest-Common-Divisor-of-Strings.cpp
class Solution
{
public:
    string gcdOfStrings(string str1, string str2) 
    {
        if (str1 + str2 != str2 + str1) 
        {
            return "";
        }

        int len1 = str1.length();
        int len2 = str2.length();

        while (len2 != 0)
         {
            int temp = len2;
            len2 = len1 % len2;
            len1 = temp;
        }

        return str1.substr(0, len1);
    }
};
